lib/rest/Twilio.ts

```diff
@@ -0,0 +1,934 @@
+'use strict'
+
+import RequestClient from '../base/RequestClient'; /* jshint ignore:line */
+import { HttpMethod } from '../interfaces'; /* jshint ignore:line */
+
+var os = require('os');  /* jshint ignore:line */
+var url = require('url');  /* jshint ignore:line */
+var moduleInfo = require('../../package.json');  /* jshint ignore:line */
+var util = require('util');  /* jshint ignore:line */
+var RestException = require('../base/RestException');  /* jshint ignore:line */
+
+interface opts {
+  httpClient?: RequestClient
+  accountSid?: string
+  env?: NodeJS.ProcessEnv
+  edge?: string
+  region?: string
+  lazyLoading?: boolean
+  logLevel?: string
+  userAgentExtensions?: string[]
+}
+
+interface requestOpts {
+  method?: HttpMethod
+  uri?: string
+  username?: string
+  password?: string
+  headers?: Headers
+  params?: object
+  data?: object
+  timeout?: number
+  allowRedirects?: boolean
+  logLevel?: string
+}
+
+/* jshint ignore:start */
+/**
+ * Parent class for Twilio Client that implements request & validation logic
+ *
+ * @constructor BaseTwilio
+ *
+ * @param {string} username -
+ *          The username used for authentication. This is normally account sid, but if using key/secret auth will be
+ *          the api key sid.
+ * @param {string} password -
+ *          The password used for authentication. This is normally auth token, but if using key/secret auth will be
+ *          the secret.
+ * @param {opts} [opts] - The options argument
+ *
+ * @returns {BaseTwilio} A new instance of BaseTwilio
+ */
+/* jshint ignore:end */
+
+class BaseTwilio {
```

Comment:
Should move this into a separate file so we all this hand-crafted code doesn't need to go in the template.